ClinicalTrial_SE_curation.Rmd

```diff
@@ -0,0 +1,320 @@
+---
+title: "Curation of Clinical trial Sumamrized Experiment objects"
+author: "Nikta Feizi, Christopher Eeles, Sisira Nair"
+date: "`r format(Sys.time(), '%a/%d/%b')`"
+output: html_document
+---
+#Please note:
+Parameters of "run_all" function are data specific and must be defined by looking at the metadata outside of the function. 
+Run "assay_data <- as.data.frame(exprs(rma.norm)) " to find "delim" which is the delimiter that separates the GSM IDs from the assay_data column names.   
+Pheno_data  column names must be used to assign column names to pre-defined columns (in "prior" vector). 
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+
+```{r}
+# configure download options for this session
+options(encoding = "UTF-8" , timeout = 1000)
+work_dir = setwd(paste0(getwd()))
+```
+
+```{r}
+# fetch Gencode v33 annotations from BHKLAB-Pachyderm/Annotations
+gencode_url <- "https://github.com/BHKLAB-Pachyderm/Annotations/raw/master/Gencode.v33.annotation.RData"
+gencode_file <- file.path(work_dir, "Ensembl.v99.annotation.RData")
+download.file(gencode_url, destfile=gencode_file)
+load("Ensembl.v99.annotation.RData")
+```
+
+
+```{r}
+#required libraries
+library(stringr)
+library(GEOquery)
+library(SummarizedExperiment)
+library(qs)
+library(affy)
+```
+
+```{r}
+geo_rma<- function (raw_tar, geo_path , brain_path){
+  
+  # Download CEL files
+  cel.file <- paste(work_dir, raw_tar, sep="/")
+  options(download.file.method="curl", download.file.extra="-k -L")
+  download.file(geo_path, cel.file)
+  
+  # Unpack the CEL files
+  exit_dir <- sub("\\..*", "", raw_tar)
+  untar(raw_tar, exdir=exit_dir)
+  cel.files<-list.files(paste(exit_dir, "/", sep=""), pattern = "gz")
+  sapply(paste(exit_dir, cel.files, sep="/"), GEOquery::gunzip)
+  
+  # Install the cdf brainarray package
+  pack_name <- tail(str_split(brain_path, "/")[[1]], n=1) 
+  cdf.file <- paste(work_dir, pack_name, sep="/")
+  options(download.file.method="curl", download.file.extra="-k -L")
+  download.file( brain_path, cdf.file)
+  install.packages(pack_name, repos = NULL, type = "source")
```

Comment:
Best Practices:

Installing packages modifies a users global state. Please always make a comment above the function definition altering a user to this.

The best way is to add a `@return` tag like:
```
#' @return None, installs packages to the global environment
installs_some_packages <- function() { ... }
```

Reply:
Done